packages/pwa-kit-react-sdk/src/ssr/universal/components/refetch-data/index.jsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import {useHistory, useLocation} from 'react-router-dom'
+import {useQueryClient} from '@tanstack/react-query'
+import LoadingSpinner from '../loading-spinner'
+
+// For good UX, show loading spinner long enough for users to see
+const LOADING_SPINNER_MIN_DURATION = 500
+
+/**
+ * Show loading spinner while refetching data on the client-side.
+ * To trigger this refetch, we do soft navigation back to the referrer.
+ * So this component is meant to be used as a route with `referrer` search param.
+ *
+ * @example
+ * const navigate = useNavigation()
+ * navigate(`/__refetch-data?referrer=${encodeURIComponent(urlOfCurrentPage)}`, 'replace')
+ *
+ * @private
+ */
+const RefetchData = () => {
+    const history = useHistory()
+    const location = useLocation()
+
+    let queryClient
+    try {
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        queryClient = useQueryClient()
+    } catch (err) {
+        // `useQueryClient` throws an error if the project does not use react-query.
+        // So in this case, leave `queryClient` as undefined. Continue to navigate to the referrer.
+    }
+
+    useEffect(() => {
+        const refetchData = async () => {
+            const showLoadingSpinner = new Promise((resolve) =>
+                setTimeout(resolve, LOADING_SPINNER_MIN_DURATION)
+            )
+            const invalidateQueries = queryClient?.invalidateQueries()
```

Comment:
I couldn't find any doc's saying that this method was `async`, but lo and behold after looking at the repo, it returns a promise. Did you have to do that too?

Reply:
It's not explicit in the docs, but it's evident from the [use of `await` in their example](https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientinvalidatequeries:~:text=tsx-,await,-queryClient.invalidateQueries).

Reply:
Ahh I see that now.. in their other doc [page](https://tanstack.com/query/v4/docs/react/guides/query-invalidation) this neglects to use `await` in their example code.